src/XmlDeserializationVisitor.php

```diff
@@ -387,4 +387,23 @@ private function getDomDocumentTypeEntitySubset($data)
 
         return $internalSubset;
     }
+
+    /**
+     * @param mixed $value
+     *
+     * @return bool
+     */
+    public function isNull($value)
+    {
+        if ($value instanceof \SimpleXMLElement) {
+            $xsiAttributes = $value->attributes('http://www.w3.org/2001/XMLSchema-instance');
+
+            //We have to keep the isset quiet, some tests give error: `Node no longer exists`; even though it evaluates to false
+            if (@isset($xsiAttributes['nil']) && (string) $xsiAttributes['nil'] === 'true') {
```

Comment:
Booleans in xml can also be represented as 1 and 0. See https://www.w3.org/TR/2004/REC-xmlschema-2-20041028/datatypes.html#boolean So this check doesn't work in this case.